barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover.hpp

```diff
@@ -54,20 +58,7 @@ template <class ProverInstances_> class ProtoGalaxyProver_ {
      * @param domain_separator  separates the same type of data coming from difference instances by instance
      * index
      */
-    void finalise_and_send_instance(std::shared_ptr<Instance>, const std::string& domain_separator);
-
-    /**
-     * @brief Execute the folding prover.
-     *
-     * @todo TODO(https://github.com/AztecProtocol/barretenberg/issues/753): fold goblin polynomials
-     * @return FoldingResult is a pair consisting of an accumulator and a folding proof, which is a proof that the
-     * accumulator was computed correctly.
-     */
-    BB_PROFILE FoldingResult<Flavor> prove();
-
-    // Returns the accumulator, which is the first element in ProverInstances. The accumulator is assumed to have the
-    // FoldingParameters set and be the result of a previous round of folding.
-    std::shared_ptr<Instance> get_accumulator() { return instances[0]; }
+    void run_oink_prover_on_instance(std::shared_ptr<Instance>, const std::string& domain_separator);
```

Comment:
it's fine if you don't want to do it in this PR but we also have an oink verifier (and recursive verifier I think) so they could be mirrored in the PG verifiers

Reply:
Yeah I'm just focusing on the prover and then I'm going to do analogous refactoring to the verifiers.

Reply:
I agree it's kinda ick but to me it feels much more efficient to work this way. Just need to make sure the verifier refactor also happens.